src/client.ts

```diff
@@ -2751,23 +2754,42 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
    * @param {boolean} options.watch Subscribes the user to the channels of the threads.
    * @param {number}  options.participant_limit Limits the number of participants returned per threads.
    * @param {number}  options.reply_limit Limits the number of replies returned per threads.
+   * @param {boolean} asJSON If set to true, returns JSON response instead of array of Thread instances.
    *
    * @returns {{ threads: Thread<StreamChatGenerics>[], next: string }} Returns the list of threads and the next cursor.
    */
-  async queryThreads(options?: QueryThreadsOptions) {
-    const opts = {
+
+  async queryThreads(
+    options: QueryThreadsOptions | undefined,
+    asJSON: true,
```

Comment:
*nitpick*
```suggestion
    asObject: true,
```
maybe something like this ? This is technically not JSON, so perhaps more fitting. Totally not important, just caught my eye

Reply:
I'll be dropping these changes altogether - I don't think we should support this at all.